src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -372,6 +374,33 @@ public static function underscore($id)
         return strtolower(preg_replace(array('/([A-Z]+)([A-Z][a-z])/', '/([a-z\d])([A-Z])/'), array('\\1_\\2', '\\1_\\2'), str_replace('_', '.', $id)));
     }
 
+    /**
+     * Fetches a variable from the environment.
+     *
+     * @param string The name of the environment variable
+     *
+     * @return scalar The value to use for the provided environment variable name
+     *
+     * @throws EnvNotFoundException When the environment variable is not found and has no default value
+     */
+    protected function getEnv($name)
```

Comment:
Right. The `.env` file will only be loaded if we are in debug mode and environment is not prod. So in prod, I specify every variable in my container configuration e.g. `docker-compose.yml`. For running unit tests or a `composer install` while building the container I do an export on the `.env.example` file which is similar to the previously `parameters.yml.dist`